docs/developers-guide/substreams-sink-files.md

```diff
@@ -0,0 +1,275 @@
+
+Verify the setup for the example project by using the `make build`, `make codegen` and `substreams run` commands.
+
+### Build the example project
+
+First, build the Substreams module by using the included `make` command.
+
+```bash
+make
+```
+
+### Generate the required code for the example
+
+Next, run the included `make codegen` command to create the required protobuf files.
+
+```bash
+make codegen
+```
+
+### Run the example
+
+Last, run the project by using the `substreams run` command.
+
+```bash
+substreams run -e mainnet.eth.streamingfast.io:443 substreams-filesink-tutorial.yaml map_eth_block_for_sink --start-block 1000000 --stop-block +1
+```
+
+### Results of example
+
+The `substreams run` command will result in output resembling the following:
+
+```bash
+----------- NEW BLOCK #1,000,000 (1000000) ---------------
+{
+  "@module": "map_eth_block_for_sink",
+  "@block": 1000000,
+  "@type": "substreams.sink.files.v1.Lines",
+  "@data": {
+    "lines": [
+      "{\"hash\":\"8e38b4dbf6b11fcc3b9dee84fb7986e29ca0a02cecd8977c161ff7333329681e\",\"number\":1000000,\"parent_hash\":\"b4fbadf8ea452b139718e2700dc1135cfc81145031c84b7ab27cd710394f7b38\",\"timestamp\":\"2016-02-13T22:54:13Z\"}"
+    ]
+  }
+}
+```
+
+## Code walkthrough
+
+Take a moment to review the various files in the accompanying Substreams project. Two important files to review include the [Substreams manifest](#) and the [`lib.rs`](#) Rust source code file that contains the module handlers for the file sink example.
+
+### Example module handler
+
+The example code in the [`lib.rs`](#) Rust source code file contains the `map_eth_block_for_sink` module handler responsible for extracting the blockchain data. The module handler is responsible for passing the data to the `substreams.sink.files.v1` protobuf for the sink tool and its processes.
+
+```rust
+#[substreams::handlers::map]
+fn map_eth_block_for_sink(block: eth::Block) -> Result<Lines, substreams::errors::Error> {
+
+    let header = block.header.as_ref().unwrap();
+
+    Ok(pb::sinkfiles::Lines {
+        lines: vec![json!({
+            "number": block.number,
+            "hash": Hex(&block.hash).to_string(),
+            "parent_hash": Hex(&header.parent_hash).to_string(),
+            "timestamp": header.timestamp.as_ref().unwrap().to_string()
+        })
+        .to_string()],
+    })
+}
+```
+
+The module handler uses `JSONL` for the output type. The [`json!`](https://docs.rs/serde_json/latest/serde_json/macro.json.html) macro is used to write the block data to the Rust `Vec` type by using the Rust [`vec!`](https://doc.rust-lang.org/std/macro.vec.html) macro.
+
+**NOTE**: <i>We just need to emphasis in the tutorial that it's an example and that a lot of "rows" can be extracted per block. Giving the ERC20/ERC721/ERC1155 examples is a good idea as we coded it: https://github.com/streamingfast/substreams-eth-token-transfers/blob/develop/src/lib.rs#L24</i>
```

Comment:
Material for section explaining the required changes to a Substreams

Reply:
Title for subsection renamed.